modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -786,7 +924,7 @@ FeatureFloodCount::visitElementalNeighbors(const Elem * elem, unsigned long curr
   std::vector<const Elem *> all_active_neighbors;
 
   // Loop over all neighbors (at the the same level as the current element)
-  for (unsigned int i = 0; i < elem->n_neighbors(); ++i)
+  for (auto end = elem->n_neighbors(), i = 0u; i < end; ++i)
```

Comment:
I agree, I'm really not liking any of this, I'm halfway tempted to go back to `unsigned int` for counters. The only problem with that is now that I'm aware of all this type flow, it bothers me just a bit that unsigned int is actually wrong for 99% of our counters. Sure it's good to about ~4 Billion but size_t is 64 bits, not 32.

Derek proposed a macro on gitter that might be worth adding to MOOSE. Did you see that?